jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/BuildImageStep.java

```diff
@@ -185,4 +187,47 @@ public DescriptorDigest getDiffId() throws LayerPropertyNotFoundException {
       return imageBuilder.build();
     }
   }
+
+  @Nullable
+  private ImmutableList<String> computeEntrypoint(
+      Image<Layer> baseImage, ContainerConfiguration containerConfiguration) {
+    boolean shouldInheritEntrypoint = containerConfiguration.getEntrypoint() == null;
+
+    if (shouldInheritEntrypoint && baseImage.getEntrypoint() != null) {
+      buildConfiguration
+          .getEventDispatcher()
+          .dispatch(
+              LogEvent.lifecycle(
+                  "Container entrypoint set to "
+                      + baseImage.getEntrypoint()
+                      + " (inherited from base image)"));
+    }
+
+    return shouldInheritEntrypoint
+        ? baseImage.getEntrypoint()
+        : containerConfiguration.getEntrypoint();
+  }
+
+  @Nullable
+  private ImmutableList<String> computeProgramArguments(
+      Image<Layer> baseImage, ContainerConfiguration containerConfiguration) {
+    boolean shouldInheritProgramArguments =
+        containerConfiguration.getEntrypoint() == null
+            && (containerConfiguration.getProgramArguments() == null
+                || containerConfiguration.getProgramArguments().isEmpty());
```

Comment:
`Strings.isNullOrEmpty` is handy here. But then, this is not consistent with ENTRYPOINT, as I said.

Reply:
The type is `ImmutableList<String>` not `String`.